### PR TITLE
Scoped user limitations

### PR DESCRIPTION
**Pull request by EgleHelms**
Limitations considered and should function:

**_When a service provider user logs into the system, the UI must be modified for them in the following aspects:_**

- +The service provider dropdown at the top of the view is removed.
- +The Settings panel includes only the Service Provider tab
- +The User dashboard allows the logged-in user to view only users associated with the current service provider
- +to create other service provider user accounts (for the same service provider) 
- +to delete accounts for the same service provider.
- +the data shown in all other panels shall be restricted to the current service provider. 
- +service provider users shall create/edit/delete new users at the account level, by selecting an existing account that falls under their service provider.

**_When an account-level user logs into the system, the UI shall be modified for them in the following aspects:_**

- +The Settings panel is removed
- +The account list is removed (since there is only one account)
- +A new Account panel is available on the left-hand side menu, it shows the current Add/Edit the current user Account panel
- +account users shall create/edit/delete new users at the account level, that falls under their service provider +account.
- +Recent Calls and Alerts are filtered to show only the calls and alerts for this account
- +The carrier panel continues to list all of the carriers defined for the account or service provider
- +Carriers can only be added at the account level.
- +only the account-level carriers can be edited or deleted.  
- +The speech panel continues to list all of the speech credentials
- +only the account level credentials can be edited or deleted.

### CODE REVIEW

**Comment by EgleHelms** [comment]
@kitajchuk this is my first try at applying the limitations according to the requirements. Will be grateful for your review.
I have a few issues in the console, regarding "icons" type and some "id" issue, I am not too sure, what changed. Maybe that is something you can take a look at.

This pull request from api-server contains the changes regarding user scopes:
https://github.com/jambonz/jambonz-api-server/pull/80

**Comment by EgleHelms** [comment]
@davehorton this is my first try to apply all the limitations as according to the requirements.
I have some issues regarding /PredefinedCarriers and /SpeechCredentials not allowing the account-scoped user to fetch or to post/put account level resources,  please check my comments there: https://github.com/jambonz/jambonz-api-server/pull/80

please feel free to just update that branch if you see an obvious mistake since the front end should work correctly after the api-server is fixed.

File: src/containers/login/create-password.tsx [hunk 1]
```diff
@@ -65,7 +72,11 @@ export const CreatePassword = () => {
           if (response.status === StatusCodes.NO_CONTENT) {
             sessionStorage.clear();
 
-            navigate(ROUTE_INTERNAL_ACCOUNTS);
+            navigate(
+              user?.scope !== USER_ACCOUNT
```
Comment (EgleHelms) [inline src/containers/login/create-password.tsx:76]: @kitajchuk this is not working, I dont know why, in the login.tsx it does.

**Comment by kitajchuk** [comment]
@EgleHelms the commit history on this branch is bizarre to me. Seems to be a ton of stuff not related to this work. I also see merge commits from main—please always rebase main into your branches and do not merge main into them. Anyway when I pull this branch locally and try to rebase main I get conflicts which should not be the case. So wondering what is going on with the timeline here on this branch?

File: src/containers/internal/navi/index.tsx [hunk 2]
```diff
@@ -172,9 +174,47 @@ export const Navi = ({
         </div>
         <div className="navi__routes">
           <ul>
-            {naviTop.map((item) => (
-              <Item key={item.label} item={item} handleMenu={handleMenu} />
-            ))}
+            {naviTop.map((item) => {
```
Comment (EgleHelms) [inline src/containers/internal/navi/index.tsx:177]: @kitajchuk this map is complaining about duplicate keys, but I cannot figure out why

**Comment by EgleHelms** [comment]
> @EgleHelms the commit history on this branch is bizarre to me. Is everything related to this work? I also see merge commits from main—please always rebase main into your branches and do not merge main into them. Anyway when I pull this branch locally and try to rebase main I get conflicts which should not be the case. So wondering what is going on with the timeline here on this branch?

Yeah, you are right, worked on things before they got merged to main.
had to cherry-pick the commits into a new branch: https://github.com/jambonz/jambonz-webapp/pull/157

**Pull request closed without merging**